playbooks/roles/ocp-install/tasks/checkcert.yaml

```diff
@@ -0,0 +1,44 @@
+---
+# tasks file for ocp4 install cert
+
+- name: check if oc is installed
+  command: type -P oc
+  register: oc_installed
+
+# check certificate validity
+# cert_valid is 1 if current time is in certificate validity time period, 0 otherwise
+- name: check the certificate validity
+  shell: |
+    dates=$(jq .ignition.security.tls.certificateAuthorities[].source openstack-upi/worker.ign |
+            sed -n -e  's/^\"data:text\/plain;charset=utf-8;base64,\(.*\)\"/\1/p' |
+            base64 --decode |
+            openssl x509 -noout -in - -dates)
+
+    notBefore=$(echo "${dates}" | sed -n -e 's/^notBefore=//p' | date -d - +%s)
+    notAfter=$(echo "${dates}" | sed -n -e 's/^notAfter=//p' | date -d - +%s)
+    present=`date +%s`
```

Comment:
This is good. @Kuldip-Nanda can you also check if you can use `checkend` option? That will do away with the sed logic
eg. `if openssl x509 -checkend 3600 -noout -in <certificate>` checks if certificate is good for 1 hr. Checking if certificate is valid for next 1 hr should be more than enough I think.

Also please raise the PR against the master/main branch.

Reply:
I know we discussed this earlier but was not sure how much time I should add.  So that is why I went to safer implementation.As you think one hour is good enough time, I can make it that.